app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MappingMissionSelectionActivity.kt

```diff
@@ -36,55 +38,131 @@ class MappingMissionSelectionActivity : AppCompatActivity() {
     @Inject
     lateinit var authService: AuthenticationService
 
-    enum class StorageType(val checked: Boolean, val sharedVisible: Boolean, val privateVisible: Boolean) {
+    enum class StorageType(
+        val checked: Boolean,
+        val sharedVisible: Boolean,
+        val privateVisible: Boolean
+    ) {
         PRIVATE(true, false, true),
         SHARED(false, true, false)
     }
 
-    private val currentType = MutableLiveData(StorageType.PRIVATE)
+    private val currentType = MutableLiveData(Pair<StorageType, String?>(StorageType.PRIVATE, null))
```

Comment:
As this is more than just the type, maybe rename the value to something more meaningful

Reply:
Done